app/src/main/java/ingsw/group1/findmyphone/PermissionInfoDialog.java

```diff
@@ -0,0 +1,128 @@
+package ingsw.group1.findmyphone;
+
+import android.app.Dialog;
+import android.content.Context;
+import android.content.DialogInterface;
+import android.os.Bundle;
+
+import androidx.annotation.NonNull;
+import androidx.appcompat.app.AlertDialog;
+import androidx.fragment.app.DialogFragment;
+
+/**
+ * Class describing a dialog that shows information related to the permissions required by
+ * the app.
+ *
+ * @author Riccardo De Zen.
+ */
+public class PermissionInfoDialog extends DialogFragment {
+
+    public static final int LOCATION = 0;
+    public static final int MESSAGES = 1;
+
+    private static final String DEFAULT_INFO = "";
+    private static final int DEFAULT_ID = 0;
+    private static final int LOCATION_INFO_STRING_ID = R.string.location_permissions_info;
+    private static final int MESSAGES_INFO_STRING_ID = R.string.message_permissions_info;
+    private static final int POSITIVE_BUTTON_ID = R.string.permissions_positve_button;
+    private static final int NEGATIVE_BUTTON_ID = R.string.permissions_negative_button;
+    private static final int TITLE_ID = R.string.permissions_info_title;
+
+    private AlertDialog.Builder builder;
+    private PermissionsDialogListener listener;
+    private int infoStringId;
+
+    /**
+     * Default constructor. Takes the type of Dialog. Only accepted values are
+     * {@link PermissionInfoDialog#LOCATION} and {@link PermissionInfoDialog#MESSAGES}. Other
+     * values
+     * will result in incorrect or empty dialogs.
+     *
+     * @param type The type of dialog. See public constants of this class.
+     */
+    public PermissionInfoDialog(int type) {
+        super();
+        switch (type) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [A switch statement does not contain a break](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=5082269)